### PR TITLE
Add docs for updated addons-manager

### LINKED ISSUE #2410: Docs for updated addons-manager
(This is used to request new product features)

**Describe the feature request**
There have been a number of designs and discussions for updated addons-manager related to the work to enable clusterclass. Add docs, overview and high level overview with docs to allow a new developer to understand and contribute 
**Describe alternatives you've considered**
N/A
**Affected product area (please put an X in all that apply)**

- [x] APIs
- [x] Addons
- [ ] CLI
- [x] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by vijaykatam**
Add docs for addons manager

Signed-off-by: Vijay Katam <vkatam@vmware.com>

### What this PR does / why we need it
Add docs, overview and high level overview with docs to allow a new developer to understand and start contributing.
### Which issue(s) this PR fixes

Fixes #2410

### Describe testing done for PR

N/A

### Release note

```release-note
package-based-lcm: Add docs for updated addons-manager
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [ ] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [ ] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: addons/README.md [hunk 1]
```diff
@@ -1,25 +1,412 @@
...
+A->>S: Update resolvedTKR in ClusterBootstrap
+A->>S: Unpause Cluster
+A-->>W: Fetch statuses of PackageInstalls
+A->>S: Update statuses in ClusterBootstrap
+```
+
+## Customizing a cluster
+While most use cases clusters should be created by just creating a Cluster object and TKG components will create an opinionated
+cluster with packages and configurations we support customizing configurations and packages. OEM distributors of TKG should
+create their own TKr package that confirms to their majority use cases so that end users need not go down the path of customizing.
```
Comment (maralavi) [inline addons/README.md:170]: ```confirms``` --> ```conforms```

File: addons/README.md [hunk 2]
```diff
@@ -1,25 +1,412 @@
...
+2. **AntreaConfig** - This is a resource that represents the configuration of antrea carvel package.
+3. **CalicoConfig** - This is a resource that represents the configuration of calico carvel package.
+4. **VSphereCPIConfig** - This is a resource that represents the configuration of vsphere cloud provider carvel package for both paravirtual and non-paravirtual modes of operation.
+5. **VSphereCSIConfig** - This is a resource that represnets the configuration of vsphere csi carvel package for both paravirtual and non-paravirtual modes of operation.
+
+## Webhooks
+1. A validating webhook for ClusterBootstrapTemplate ensures that
+   1. it can only be created in a system namespace
+   2. required packages such as kapp and cni are set
+   3. only type of valuesFrom is defined for a package
```
Comment (maralavi) [inline addons/README.md:32]: should be: only ```one``` type?

File: addons/README.md [hunk 3]
```diff
@@ -1,25 +1,412 @@
...
+   providerRef:
+      apiGroup: cni.tanzu.vmware.com
+      kind: CalicoConfig
+      name: sample-config
+   secretRef: my-secret-key
+   inline:
+      key1: value1
+      nested_key: 
+        key2: value2
+```
```
Comment (maralavi) [inline addons/README.md:299]: Thought: Though it is mentioned in the paragraph above, isn't it confusing that the example lists all ```valuesFrom```  options altogether? Wouldn't it be better if this was in the form of 3 different examples?

File: addons/README.md (hunk 1 above)
Comment (vijaykatam) [reply addons/README.md]: fixed

File: addons/README.md (hunk 2 above)
Comment (vijaykatam) [reply addons/README.md]: yup, fixed

File: addons/README.md (hunk 3 above)
Comment (vijaykatam) [reply addons/README.md:323]: Good point, added 3 separate examples

File: addons/README.md [hunk 4]
```diff
@@ -1,25 +1,412 @@
...
+The mechanism for determining the versions of packages are determined by an imgpkg bundle that contains a number of resources which are described below.
 
-## Watch
+## APIs
 
-Tanzu Addons Manager watches the following
+The following are the primary APIs used by the addons manager for determining what packages, versions and configurations go into a kubernetes cluster.
+
+1. **TanzuKubernetesRelease(tkr)** - This is a resource that represents a summary of items contained in a Tanzu Kubernetes release. 
+It aggregates VM images using OSImages Resource, packages and versions shipped with a release. It specifies all the components
```
Comment (maralavi) [inline addons/README.md]: ```OSImages Resource```-->```OSImage resource```

File: addons/README.md [hunk 5]
```diff
@@ -1,25 +1,412 @@
...
+2. A validating webhook for ClusterBootstrap ensures that
+   1. required packages such as kapp and cni are set and valid
+   2. any package being set is available in a system namespace(provided by TKR components)
+   3. an update does not remove required packages
+   4. an update does not downgrade a package version
+   5. an update does not delete any package
+   6. an update does not change package ref for core packages, for example changing cni from antrea to calico is not allowed
+   7. an update does not alter the mechanism used to provide valuesFrom
+3. A defaulting webhook for ClusterBootstrap is a convenience to enable API users to provide partial objects with an annotation 
+to fill missing information from a ClusterBootstrapTemplate. This allows a user to customize packages or specific package in a cluster.
```
Comment (maralavi) [inline addons/README.md:44]: nit: ```packages or specific package``` --> ```packages or a specific package```

File: addons/README.md [hunk 6]
```diff
@@ -1,25 +1,412 @@
...
+A->>W: Update PackageInstalls for others with nodeSelector for workload clusters on supervisor
+A->>S: Update resolvedTKR in ClusterBootstrap
+A->>S: Unpause Cluster
+A-->>W: Fetch statuses of PackageInstalls
+A->>S: Update statuses in ClusterBootstrap
+```
+
+## Customizing a cluster
+While most use cases clusters should be created by just creating a Cluster object and TKG components will create an opinionated
+cluster with packages and configurations we support customizing configurations and packages. OEM distributors of TKG should
```
Comment (maralavi) [inline addons/README.md:177]: ```While most use cases``` --> ```While in most use cases```

Comment (maralavi) [reply addons/README.md:177]: Also I think its better to add a comma ```,``` before the phrase starting with ```we support```

File: addons/README.md [hunk 7]
```diff
@@ -1,25 +1,412 @@
...
+      kind: VSphereCluster
+   clusterNetwork:
+      pods:
+         cidrBlocks: [ "192.168.0.0/16","fd00:100:96::/48" ]
+      services:
+         cidrBlocks: [ "192.168.0.0/16","fd00:100:96::/48" ]
+...
+...
+```
+When the config resource is pre-created addons-controller will use it in lieu of the one provided by CBT.
```
Comment (maralavi) [inline addons/README.md:216]: I think a comma ```,``` is needed after ```is pre-created```

**Comment by maralavi** [comment]
LGTM, just some minor nit comments

**Review by maralavi — APPROVED** [review]

**Comment by maralavi** [comment]
Thanks for the PR, LGTM

**Pull request merged**